app/rubin-2025/register-form.jsx

```diff
@@ -79,233 +79,294 @@ class RegisterForm extends React.Component {
+          <span className="checkbox-block">
+            <input
+              checked={this.state.underAge}
+              disabled={inputDisabled}
+              id="register-form-underAge"
+              name="underAge"
+              onChange={this.updateAge}
+              type="checkbox"
+            />
+            <label htmlFor="register-form-underAge">
+              <Translate component="span" content="registerForm.underAge" />
+            </label>
+          </span>
+        </div>
+
+        <div className="form-row">
+          <div className="label-block">
+            <label htmlFor="register-form-login">
+              <Translate content="registerForm.userName" />
+            </label>
+
+            <div aria-live="polite" id="register-form-login-help-message-1">
+              {badNameChars?.length > 0 && (
+                <Translate className="form-help error" content="registerForm.badChars" />
+              )}
+            </div>
+
             {"nameConflict" in this.state.pending && (
               <LoadingIndicator />
             )}
-            {nameConflict && (
-              <span className="form-help error">
-                <Translate content="registerForm.nameConflict" />{' '}
-                <a href={`${window.location.origin}/reset-password`}>
-                  <Translate content="registerForm.forgotPassword" />
-                </a>
-              </span>
-            )}
+
+            <div aria-live="polite" id="register-form-login-help-message-2">
+              {nameConflict && (
+                <span className="form-help error">
+                  <Translate content="registerForm.nameConflict" />{' '}
+                  <a href={`${window.location.origin}/reset-password`}>
+                    <Translate content="registerForm.forgotPassword" />
+                  </a>
+                </span>
+              )}
+            </div>
+
             {(this.state.input_login?.length > 0
               && nameConflict === false
+              && !("nameConflict" in this.state.pending)
             ) && (
               <span className="form-help success">
                 <Translate content="registerForm.looksGood" />
               </span>
             )}
+
             <Translate className="form-help info right-align" content="registerForm.required" />
-          </span>
+          </div>
+
           <input
-            type="text"
-            name="login"
-            id="register-form-login"
-            value={this.state.input_login}
+            aria-describedby="register-form-error-message register-form-login-help-message-1 register-form-login-help-message-2"
             className="standard-input full"
             disabled={inputDisabled}
-            autoFocus
-            onChange={this.handleUserInput}
+            id="register-form-login"
             maxLength="255"
+            name="login"
+            onChange={this.handleUserInput}
+            type="text"
```

Comment:
```suggestion
            type="text"
            autocomplete="username"
```

[Add autocomplete to input fields](https://www.w3.org/WAI/WCAG21/Techniques/html/H98) to meet WCAG SC 1.3.5.